### ISSUE TITLE
We should add implicit casts from string, i.e. to pool

### ISSUE THREAD

**Issue by premun** [sharpliner/sharpliner#203]
It would be nice to have this:
```csharp
	Pool = "windows-latest"
```
as a shorthand for
```csharp
	Pool = new HostedPool(vmImage: "windows-latest")
```

**Comment by AHeiming** [comment]
Hello,

I would like to work on this, if this is okay for you.

**Comment by premun** [comment]
Absolutely! Thanks!

Please give it some thought, I wasn't especially verbose with the issue as it occurred to me briefly today.
I think vmImage makes the most sense but maybe we could also set a tuple and have the pool name too? Totally up to you, what you think is useful

**Comment by premun** [comment]
Ah, sorry seems like homepage lost the "How to develop".

The CI project is a bit tricky and you can either unload it and work without it or you need to `dotnet build` the CI project from console.
You can then dotnet clean and build it in case you want to see the latest changes in the project.

What it does is build a package of the 43 version locally and the CI project loads it like it would get it from nuget.org to E2E test it. You can then also test the new feature there if you clean/build. You can also test it inside of the unit test project which has a direct reference.

Let me know if you have any other problems

**Comment by AHeiming** [comment]
Hi thanks for the help, it works now.
Regarding the issue:

I think there might be some confusion if we implement this.
`Pool` is the base class of `HostedPool`. So every time a `Pool` is demanded, it does not necessarily has to be an instance of `HostedPool`.

My current approach was using the implicit operator to accomplish this.
But if we do this, everytime in the code a `Pool ` is instantiated without a constructor it will be of type `HostedPool`. Is this the desired behaivor?

I don't know how often `HostedPool` is chosen over the other Pools, I just want to avoid confusion.
This is my current approach:

`public static implicit operator Pool(string vmImage)
    {
        return new HostedPool(vmImage: vmImage);
    }`


I hope this was understandable 😄

**Comment by premun** [comment]
Yes, I think that's roughly what I thought it should be. It's quite simple I guess but if I'm not wrong the YAML would then look almost the same and mean the same?

**Comment by AHeiming** [comment]
Okay, I just wanted to be sure that this is the desired behaivour. 😄 
And yes, regarding the YAML file, I think so. 
I hope to be able to send a pull request today/tomorrow

**Comment by premun** [comment]
Great! I'll be looking forward to it

**Comment by premun** [comment]
Resolved by #209
